pkgdb/src/buildenv/realise.cc

```diff
@@ -260,20 +261,27 @@ getLockedPackages( resolver::Lockfile & lockfile, const System & system )
  * message).
  * @return List of realised packages and their store paths for referencing.
  * @throws PackageEvalFailure if the package fails to evaluate.
+ * @throws PackageEvalFailure if the package is marked as broken or unfree and
+ * neither is allowed through the options.
  * @throws PackageBuildFailure if the package fails to build.
  * @throws PackageUnsupportedSystem if the package is not available for the
  given system.
  */
 static std::vector<std::pair<buildenv::RealisedPackage, nix::StorePath>>
-getRealisedPackages( nix::EvalState &                   state,
-                     const std::string &                pId,
-                     const resolver::LockedPackageRaw & package,
-                     const System &                     system )
+getRealisedPackages( nix::EvalState &                        state,
+                     const std::string &                     pId,
+                     const resolver::LockedPackageRaw &      package,
+                     const System &                          system,
+                     const flox::resolver::Options::Allows & allows )
 {
   std::vector<std::pair<buildenv::RealisedPackage, nix::StorePath>> realised;
 
 
-  auto packageInputRef = nix::FlakeRef( package.input );
+  /* Ensure the input is fetched with `flox-nixpkgs `*/
```

Comment:
nonblocking: add note/todo that as soon as we support inputs other than github:nixos/nixpkgs, we'll need to change this so it doesn't throw an exception